src/instructlab/configuration.py

```diff
@@ -586,13 +793,209 @@ def ensure_storage_directories_exist():
+            "learning_rate": 2e-6,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+        },
+        ckpt_output_dir="~/.local/share/instructlab/checkpoints",
+        data_output_dir="~/.local/share/instructlab/internal",
+        data_path="~/.local/share/instructlab/datasets",
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=False,
+        is_padding_free=True,
+        model_path="~/.cache/instructlab/models/granite-7b-starter",
+    )
+
+    FOUR_L_FORTY_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-6,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+            "deepspeed_cpu_offload_optimizer_ratio": 1,
+            "deepspeed_cpu_offload_optimizer_pin_memory": False,
+        },
+        ckpt_output_dir="~/.local/share/instructlab/checkpoints",
+        data_output_dir="~/.local/share/instructlab/internal",
+        data_path="~/.local/share/instructlab/datasets",
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=True,
+        is_padding_free=True,
+        model_path="~/.cache/instructlab/models/granite-7b-starter",
+    )
+
+    EIGHT_L_FOUR_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-6,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+            "deepspeed_cpu_offload_optimizer_ratio": 1,
+            "deepspeed_cpu_offload_optimizer_pin_memory": False,
+        },
+        ckpt_output_dir="~/.local/share/instructlab/checkpoints",
+        data_output_dir="~/.local/share/instructlab/internal",
+        data_path="~/.local/share/instructlab/datasets",
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=True,
+        is_padding_free=True,
+        model_path="~/.cache/instructlab/models/granite-7b-starter",
+    )
+
     # create exper_args file for users to see/edit
     if not os.path.isfile(DEFAULTS.TRAIN_ADDITIONAL_OPTIONS_FILE):
         with open(
             DEFAULTS.TRAIN_ADDITIONAL_OPTIONS_FILE, "w", encoding="utf-8"
         ) as outfile:
             yaml.dump(additional_args_and_defaults, outfile)
 
+    if not os.path.isfile(DEFAULTS.INTERNAL_CONFIG_FILE):
+        with open(DEFAULTS.INTERNAL_CONFIG_FILE, "w", encoding="utf-8") as outfile:
+            yaml.dump(get_dict(internal_config), outfile)
+
+    if not os.path.isfile(DEFAULTS.TRAIN_A100_H100_X8_PROFILE):
```

Comment:
Seems very repetitive. Probably a `for` loop would be neater.

Reply:
good point, I will do that